compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -245,17 +246,24 @@ object RefChecks {
           (if (otherAccess == "") "public" else "at least " + otherAccess))
       }
 
-      def compatibleTypes =
-        if (member.isType) { // intersection of bounds to refined types must be nonempty
-          member.is(BaseTypeArg) ||
-          (memberTp frozen_<:< otherTp) || {
-            val jointBounds = (memberTp.bounds & otherTp.bounds).bounds
-            jointBounds.lo frozen_<:< jointBounds.hi
+      def compatibleTypes(memberTp: Type, otherTp: Type): Boolean =
```

Comment:
I suggest using other names than `memberTp` and `otherTp` here, because `memberTp` and `otherTp` are already in scope at this point and they could easily be confused when reading the code.